### ISSUE TITLE
Merge down passwd users sections in appended configs

### ISSUE THREAD

**Issue by arithx** [coreos/ignition#719]
# Feature Request #

## Desired Feature ##

When configs are appended together merge down the `users` entries in `passwd`. Currently the final config that is produced is unintuitive and can have some weird outcomes as a result of #608 

Example input config A:
```
{
  "ignition": {
    "version": "3.0.0-experimental",
    "config": {
      "append": [
        {
          "source": "https://foo/bar/baz/b"
        }
      ]
    }
  },
  "passwd": {
    "users": [
      {
         "name": "core",
         "sshAuthorizedKeys": [
           "sshKeyA"
         ],
         "homeDir": "/home/a"
      }
    ]
  }
}
```

Example input config B:
```
{
  "ignition": {
    "version": "3.0.0-experimental"
  },
  "passwd": {
    "users": [
      {
         "name": "core",
         "sshAuthorizedKeys": [
           "sshKeyA",
           "sshKeyB"
         ],
         "homeDir": "/home/b"
      }
    ]
  }
}
```

The resulting config will currently create `/home/a`, `/home/b` and the final config will look like something like:
```
{
  "ignition": {
    "version": "3.0.0-experimental"
  },
  "passwd": {
    "users": [
      {
         "name": "core",
         "sshAuthorizedKeys": [
           "sshKeyA"
         ],
         "homeDir": "/home/a"
      },
      {
         "name": "core",
         "sshAuthorizedKeys": [
           "sshKeyA",
           "sshKeyB"
         ],
         "homeDir": "/home/b"
      }
    ]
  }
}
```

**Comment by ajeddeloh** [comment]
Solving this one case only is an ugly, backwards incompatible hack. A patch fixing this should not be applied to the `spec2x` branch since it might break CL users.

Solving this in the general case is a large, backwards incompatible refactor that we're doing to spec 3.0.0 (https://github.com/coreos/ignition/pull/720 is the start of that).

If OpenShift wants this, I'd be willing to write/help write a patch so that they can carry and apply during the Ignition RPM build, but I don't want to carry it upstream if I can avoid it.

**Comment by ajeddeloh** [comment]
Closed via #741
